nav2_collision_monitor/src/polygon.cpp

```diff
@@ -392,6 +433,30 @@ bool Polygon::getCommonParameters(
           node->get_parameter(polygon_name_ + ".footprint_topic").as_string();
       }
     }
+
+    // By default, use all observation sources for polygon
+    const std::vector<std::string> default_observation_sources;
+    nav2_util::declare_parameter_if_not_declared(
+      node, "observation_sources", rclcpp::ParameterValue(default_observation_sources));
+    const std::vector<std::string> observation_sources =
+      node->get_parameter("observation_sources").as_string_array();
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".sources_names", rclcpp::ParameterValue(observation_sources));
+    sources_names_ = node->get_parameter(polygon_name_ + ".sources_names").as_string_array();
+
+    // Check the observation sources configured for polygon are defined
+    for (auto source_name : sources_names_) {
+      if (std::find(observation_sources.begin(), observation_sources.end(), source_name) ==
+        observation_sources.end())
+      {
+        RCLCPP_ERROR_STREAM(
+          logger_,
+          "Observation source [" << source_name <<
+            "] configured for polygon [" << getName() <<
+            "] is not defined!");
```

Comment:
```suggestion
            "] is not defined as one of the node's observation_source!");
```

Reply:
^ mostly just for clarify for users about how to fix

Reply:
Done :)